xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authorization/xwiki-platform-security-authorization-api/src/main/java/org/xwiki/security/authorization/Right.java

```diff
@@ -285,6 +295,32 @@ private Right(String name, RuleState defaultState, RuleState tieResolutionPolicy
         }
     }
 
+    /**
+     * @param neededRights the rights required for this right to be granted
+     * @since 15.10.2
+     */
+    @Unstable
+    public void setNeededRights(Set<Right> neededRights)
```

Comment:
I find it a bit weird that it's a setter and not part of the constructor. What does it mean to call that method after the rights are computed? E.g. I could have in theory a component performing `Right.EDIT.setNeededRights(Set.of())` at any moment and calling a cache invalidation to recompute the rights: would that have an impact? 
Looks like the set of needed rights should be final no?

Reply:
I'm also really not a fan of making Right publicly mutable, especially when I don't see any use case for it.

Reply:
> My point was to not add another constructor parameter, in particular as we already have too many.

I agree that it's not nice to add another parameter to the constructor: what we can maybe do is adding a new constructor with a map String object which would contain all properties that are not mandatory.

Reply:
Usually, a constructor becoming too big is when RightBuilder tends to appear.